migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
+    mine_report_submission_status_code    varchar(3)                              NOT NULL            ,
+    active_ind                            boolean DEFAULT true                    NOT NULL            ,
+    create_user                           character varying(60)                   NOT NULL            ,
+    create_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+    update_user                           character varying(60)                   NOT NULL            ,
+    update_timestamp                      timestamp with time zone DEFAULT now()  NOT NULL            ,
+
+    FOREIGN KEY (mine_report_id) REFERENCES mine_report(mine_report_id) DEFERRABLE INITIALLY DEFERRED ,
+    FOREIGN KEY (mine_report_submission_status_code) REFERENCES mine_report_submission_status_code(mine_report_submission_status_code) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_submission is 'A file submission for a code required report.';
+ALTER TABLE mine_report_submission OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_document_xref
+(
+    mine_report_document_xref_guid uuid    DEFAULT gen_random_uuid() NOT NULL PRIMARY KEY,
+    mine_document_guid             uuid                              NOT NULL            ,
+    mine_report_submission_id      integer                           NOT NULL            ,
+
+    FOREIGN KEY (mine_document_guid) REFERENCES mine_document(mine_document_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_submission_id) REFERENCES mine_report_submission(mine_report_submission_id) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_document_xref is 'Links a mine document to report submission.';
+ALTER TABLE mine_report_document_xref OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_comment (
+    mine_report_comment_id           serial                                  NOT NULL PRIMARY KEY,
+    mine_report_comment_guid         uuid                                    NOT NULL            ,
+    mine_report_id                   integer                                 NOT NULL            ,
+    mine_report_submission_id        integer                                                     ,
+    comment                          varchar(300)                            NOT NULL            ,
+    minespace_user_id                integer                                                     ,
+    core_user_id                     integer                                                     ,
+    comment_visibility               boolean                                 NOT NULL            ,
```

Comment:
May want to name this column "comment_visible_ind". The "ind" keeps the boolean type consistent with most other booleans in the model. And because it sounds like there is some business logic applied to this column, there should probably be a column comment to describe what this column means.

Reply:
Also may want a default value.

Reply:
changed and added a default.